### ISSUE TITLE
Add serializeUintQuantity cheatcode

### ISSUE THREAD

**Issue by mothran** [foundry-rs/foundry#7758]
### Component

Forge

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (63fff35 2024-04-22T00:18:27.941506275Z)

### What command(s) is the bug in?

forge script

### Operating System

Linux

### Describe the bug

When using `serializeUint` to serialize out deployment parameters (wallet address + private key) for end to end testing I noticed when serializing uint256's (from `makeAddrAndKey`) that the JSON output would be invalid and hard to parse with standard tools because `serializeUint` emits normal JSON integers instead of string-ifed integers in hex or integer representation. Because javascript's (and thus lots of JSON parsers) max int is 2^53 full uint256's can't be held within a normal integer type. 

Example output from `serializeUint` to json:

```
  "test_key": 53286107159197333440670927965046399095550357839745059449004494585891422388798,
```

It would be nice if this was in hex form in a string (like how `serializeAddress` emits things).

**Comment by mattsse** [comment]
I see,
we still want a way to serialize numbers,
perhaps we should add `serializeQuantity/Hex` that serializes them as hex?

and consider renaming serializeUint into serializeNumber

**Comment by mothran** [comment]
Yeah that would make more sense in my mind, the easy work around is something like:

```
        vm.serializeBytes32(obj, "test_key", bytes32(test_key));
```

So might not be as much of a pain, but still worth considering.

**Comment by mattsse** [comment]
wdyt @mds1 @DaniPopes ?

imo large uint should be serialized as quantity (hex: 0x..) because even stringified numbers are not ideal

**Comment by yash-atreya** [comment]
> wdyt @mds1 @DaniPopes ?
> 
> imo large uint should be serialized as quantity (hex: 0x..) because even stringified numbers are not ideal

+1 on serializing large uints as hex.

**Comment by kamuik16** [comment]
> wdyt @mds1 @DaniPopes ?
> 
> imo large uint should be serialized as quantity (hex: 0x..) because even stringified numbers are not ideal

I can take this, if everyone is on board.

**Comment by mattsse** [comment]
assigned
